### ISSUE TITLE
FPS drop when running on sandboxie

### ISSUE THREAD

**Issue by ccaioccesar** [sandboxie-plus/Sandboxie#1937]
### What happened?

I'm running Sandboxie inside a `hyper-v` machine, I `did GPU-Passthrough` from the host (win10) to the machine (win10).

The machine and apps recognized the GPU, when they are running `outside` of the Sandboxie everything works smoothly and correctly, however, when I run the apps inside of Sandboxie, it looks like Sandboxie blocks them from accessing the GPU, fps drops by half and the graphics look different, like if there's no GPU available.

There's an option that I could be enabling/disabling to help solve this issue?

### Download link

Not relevant

### Expected behavior

GPU detected, no fps drop

### What is your Windows edition and version?

Windows 10 version 10.0.1.19044

### In which Windows account you have this problem?

User account with UAC protection completely turned off.

### Please mention any installed security software

Only windows defender

### What version of Sandboxie are you running?

1.0.22

### In which sandbox type you have this problem?

In a Standard isolation sandbox (yellow sandbox icon).

### Where is the program located?

The program is installed only outside the sandbox.

### Can you reproduce this problem on an empty sandbox?

I can confirm it also on an empty sandbox.

### Did you previously enable some security policy settings outside Sandboxie?

No, its a clean machined, there's nothing installed, only sandboxie

**Comment by DavidXanatos** [comment]
yes you could run the apps in a compartment mode style box as this has much less isolation

**Comment by ccaioccesar** [comment]
@DavidXanatos What's a compartment mode style box?

**Comment by ccaioccesar** [comment]
@DavidXanatos even running inside a green box, it still blocking the GPU, as the graphics look different when running outside/inside, also fps drops by half.

**Comment by DavidXanatos** [comment]
Then you need to open some additional resources, if you provide a download link for one of the affected apps together with a guide how to distinguisch correct from fault operation i can take a look and provide more guidence

**Comment by ccaioccesar** [comment]
This is the app: https://www.pokexgames.com/

Running the app **outside** the `hyper v machine`
~500fps outside Sandboxie
~450fps inside Sandboxie and the graphics remain the same thing.

Running the app **inside** of the `hyper v machine`
~500fps outside Sandboxie (GPU recognized) 
~100fps inside Sandboxie, no GPU, graphics look very weird

**Comment by Cyberforensix** [comment]
Sorry for just chipping in here, but i might be able too assist some on this since it seems familiar to me, and i suspect the root cause here might have very little to do with Sandboxie\s behavior.

I strongly believe this to be related to one of the below things as i have had many similar issues and dating back many years with sandboxie in virtualized environments.  

Another reason why i strongly believe so is because i can both reproduce and mitigate (or workaround) almost identical issue myself

First and easiest/quickest thing to establish is if the app/binary/game or whatever payload it is requires more direct HW / io access through paravirtualization then just GPU-Paravirtualization.

1. Enabling and allowing FULL para-virtualization within the Hyper-V guest. The ability to enable and use this feature is dependant on your hardware supporting it and all prerequisites but usually also requires some manual configuration changes (from poweshell).

Since you are able to share the host GPU it means that you have already enabled WDDM GPU Paravirtualization or GPU-PV in short which is GPU paravirtualization, my guess is the rest of your hardware is probably supported for this. 

Under some contexts some GPU accelerated apps or games just wont work correctly or execute at all under the circumstances you have described when also 'sandboxied'.  

Ive ran into this and/or many and very similar issues  many times as i work a lot with many different applications that are GPU accelerated, i also happen to work a lot in and with virtualized infrastructures, and finally add to that being an avid sandboxie user for ages.

The second suggestion is to dig into WDAG/VBS settings.

2 .Usually from my experience, when it concerns Hyper-V these issues are related to VM security settings and policies for cpu executions inside the vm which commonly will _silently_ (yes, need debugging level to show in logs often) kill or abort the payload being executed.

These are sometimes tricky to figure out and will vary between and from different programs/games and configurations. Or, requires full VM paravirtualization (at least for CPU in addition to the GPU not sure how important full SR-IOV will be in all diffrent/to many possible scenarios) . Since you are running this setup and being a sandboxie user i take you for being 'techie' enough to know your things and basics so google will be your friend here and will guide you to the right places these are all well documented and actively discussed topics). 

If the above does not seem to have anything to do with your specific issue then i would bet my dimes on HEVCI/WDAG and VBS/Core Isolation interfering with your life. 

Solving can be tricky sometimes but often solved from first troubleshooting by whitelisting the binary payload and path from the various (sometimes all of them) memory protection settings in VBS/Core Isolation or disable VBS/Core isolation entirely. Needless to say but you will be removing most security features for this VM while doing so. And on a sidenote, in your case and setup any VBS/Core isolation whitelistings or exclusions you make have to be applied to the sandboxie binaries as well. Again, googling the terms used here will quickly guide you to the right resources.

Hope this can aid you in getting this issue solved.
